### ISSUE TITLE
Error when passing object

### ISSUE THREAD

**Issue by OmgImAlexis** [anywhichway/txi#1]
Not quite sure what's causing this or where to even start looking. Any help would be appreciated.

```
TypeError: Cannot assign to read only property 'name' of function 'async function* () {
                                let i = 0,
                                        key = keys[i];
                                while(key) {
                                        yield key;
                                        key = keys[++i];
                                }
                        }'
    at Txi.index (/Users/xo/code/vault/vault.wtf/node_modules/txi/index.js:434:18)
```

**Comment by anywhichway** [comment]
@OmgImAlexis I took a look at your commit, but there are so many line changes I can't figure out what the root cause of the issue was or how you fixed it. Can you elaborate?

Thanks for your use and support of the package.

**Comment by anywhichway** [comment]
@OmgImAlexis I looked further and found that what would explain the issue is an object being passed in for storage that also happens to be a function, since function objects can't have their name property changed. See where the 'keys' variable is set below which is then used on line 434. Not sure if this is what you found.

```
function Txi({stops,stems=true,trigrams=true,compressions=true,misspellings=true,onchange,storage={}}={}) {
		const defaults = {stops,stems,trigrams,compressions,misspellings,onchange,storage};
		// ensure Txi is called with the new operator
		if(!this || !(this instanceof Txi)) {
			return new Txi(defaults);
		}
		let {get,set,keys,count} = storage;
```

**Comment by OmgImAlexis** [comment]
Yeah I noticed that after the fact.

The reason I had to make so many changes is there was no linting run from what I can tell so there were quite a few small "issues" that I found.

I also had to remove the wrapper(which isn't recommended at all anymore) as it didn't allow intellisense to work. A better way to handle this is to have the user use babel, etc. If you **REALLY** want to also supply a out of the box version for users then you're better off publishing 2 files to npm. One that's for node and one that's for browsers.

**Comment by OmgImAlexis** [comment]
By the way this is an easier look at the commit. Whitespace changes took up a lot of the commit.

https://github.com/OmgImAlexis/txi/commit/e9fb8a99b71b3e89be0bd3b439e7552a5457ad90?w=1

**Comment by anywhichway** [comment]
@OmgImAlexis Yeah I understand on the wrapper thing. I should use Babel more; however, I have had two situations when Babel actually broke code ... even stymied some Babel developers. And Babel usually makes the browser code larger than it needs to be. I simply try and write code that is isomorphic day one and generally don't find it takes longer for small efforts so long as I ignore IE as a target.

With respect to your changes it looks like they are mostly stylistic, e.g. spacing, some var replacements, singular variable declarations rather than series, replacing some functions with in-lines, etc. Were there actually any bugs you found and fixed?

Happy to take a pull request on this with improved code, better build process, etc. and collaborate a little. It has not been a major focus.
